### PR TITLE
Supriya | MOBN-1875 | Add support to make non-default fields to be mandatory in add/edit appointments 

### PR DESCRIPTION
**Pull request by supriyatw**

### CODE REVIEW

File: ui/react-components/components/AddAppointment/AddAppointment.jsx [hunk 1]
```diff
@@ -177,12 +178,22 @@ const AddAppointment = props => {
         return {...prevAppointmentDetails, ...modifiedAppointmentDetails}
     });
 
+    const isMandatoryAndEmpty = (fieldName, value) =>{ return isMandatory(appConfig, fieldName) ? _.isEmpty(value) : false };
+
     const isValidAppointment = () => {
         const isValidPatient = appointmentDetails.patient && appointmentDetails.patient.value.uuid;
-        const startTimeBeforeEndTime = isStartTimeBeforeEndTime(appointmentDetails.startTime, appointmentDetails.endTime);
+        const startTimeBeforeEndTime = isStartTimeBeforeEndTime(appointmentDetails.startTime, appointmentDetails.endTime);;
```
Comment (binduak) [inline ui/react-components/components/AddAppointment/AddAppointment.jsx]: @supriyatw please remove extra semi-colon here

File: ui/react-components/components/AppointmentEditorCommonFieldsWrapper/AppointmentEditorCommonFieldsWrapper.jsx [hunk 2]
```diff
@@ -58,6 +63,14 @@ const AppointmentEditorCommonFieldsWrapper = props => {
...
     };
 
+    const getProviderErrorMessages = () => {
+        if (errors.providerError) {
+            return getMaxAppointmentProvidersErrorMessage(intl,
+                appConfig && appConfig.maxAppointmentProviders || DEFAULT_MAX_APPOINTMENT_PROVIDERS).providerErrorMessage;
+        }
+        if (errors.providerMandatoryError) return errorTranslations.providerMandatoryErrorMessage;
+    };
+
```
Comment (binduak) [inline ui/react-components/components/AppointmentEditorCommonFieldsWrapper/AppointmentEditorCommonFieldsWrapper.jsx:73]: @supriyatw is this a new error that we are handling as part of this piece of code ?

Comment (supriyatw) [reply ui/react-components/components/AppointmentEditorCommonFieldsWrapper/AppointmentEditorCommonFieldsWrapper.jsx:73]: yes @binduak there should be two errors 1. if the providers limit exceeds the max number of providers given in config  2. if the provider is mandatory and empty

Comment (binduak) [reply ui/react-components/components/AppointmentEditorCommonFieldsWrapper/AppointmentEditorCommonFieldsWrapper.jsx:73]: sure..this was missing and you have handled it in the code

**Review by binduak — APPROVED** [review]

**Pull request merged**